poethepoet/task/base.py

```diff
@@ -136,7 +136,7 @@ def run(
         """
         Run this task
         """
-        env = dict(env or {}, **self._config.global_env)
+        env = dict(env or {}, **self._config.global_env, **self._config.envfile)
```

Comment:
I believe this will blow up if the same variable is defined both in the global options and in the dotenv file.

Reply:
No, one of them will overwrite. Not sure what you mean by "blow up"

Reply:
```python
>>> dict({}, **({"foo": "bar"}), **({"foo": "baz"}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: type object got multiple values for keyword argument 'foo'
```
Using the update method would be more robust. That said I think it might more sense for the more local variables (in the global options) to take precedence.

Reply:
>  That said I think it might more sense for the more local variables (in the global options) to take precedence.

You mean,

envfile overwrites local overwrites global

```
envfile > local > global
```

Reply:
I mean `self._config.global_env` overwrites `self._config.envfile`. So the most explicit/local source takes precedence.

Reply:
There will be global envfile and normal envfile? Just treat it like env but there will be parsing, etc.

In other words, the implementation should be similar to env except

1. it’s not a dict; it's a string signifying the path to the envfile
2. Parsing will be done.

Reply:
In case it isn't clear, `self._config.global_env` is populated in the toml file like so:

```toml
[tool.poe.env]
VAR1 = "foo"
VAR2 = "foo"
```

These values should override any external dotenv file, not the other way around.

hence:
```python
env = dict(env or {})
env.update(self._config.envfile)
env.update(self._config.global_env)
```